docs/design/copy-tracking.md

```diff
@@ -10,17 +10,686 @@ results of rebasing across remote copies.
+in the destination tree. For example, if the tree conflict is `A+(B-C)+(D-E)`,
+then we will rewrite the `(B-C)` diff and the `(D-E)` diff to the paths in `A`.
+To translate the `(B-C)` diff, we calculate renames from `C` to `A` and then we
+apply those renames to both `C` and `B`. This may result in conflicts.
+
+If a file has a conflict in the copy ID, it will appear as if doesn't exist when
+materialized. It will therefore not show up the working copy until the user has
+resolved the conflict.
+
+Should we propagate changes to copies? For example, if you've modified file
+`foo` and then rebase it onto a commit that copied `foo` to `bar`, should we
+apply your change to `bar` too? Mercurial does that but Git doesn't. It's
+particularly useful when a file has been split in two. For example, let's say
+you've made various changes in file `foo` and then rebase those change onto a
+commit that split `foo` into `foo1` and `foo2` (or `foo` and `bar`). If we
+propagate the changes to both files, each change will apply successfully in one
+file (assuming the changes do not overlap with the split boundary). Each change
+will have a modify/delete conflicts in the other file. Those can relatively
+easily be resolved in favor of the deleted hunk. If we do not propagate changes,
+then changes that belong in one of the files will instead only appear as
+modify/delete conflicts in the first file and you will have to manually copy
+over the changes to the copied file.
+
+Propagating changes to copies means that rebasing a commit and then rebasing it
+back is no longer a no-op even when ignoring the "same-change rule". For
+example, if your commit modifies file `foo` and you rebase that commit onto a
+commit that copied `foo` to `bar`, and then you rebase it back, the same change
+will be applied twice to `foo`. However, thanks to the same-change rule, we
+won't consider it a conflict, so maybe it actually works well in practice.
+
+For example:
+```
+M set foo="bye"
+|
+| L rename foo->bar
+|/
+K add foo="hello"
+```
+
+When rebasing `L` onto `M`,
+
+When rebasing `M` onto `L`, we apply the `foo->bar` rename to the trees in `M`
+and its parent.
+
+
+```
+N rename foo->bar
+|
+| M create foo="M"
+| |
+| L delete foo
+|/
+K add foo="K"
+```
+
+When rebasing `M` onto `N`, we find the `foo->bar` rename in `N`, but since it
+is unrelated to the `foo` file in `M` (assuming the `foo` file created in `M`
+used a different salt), we will not perform any renames. The new `foo` file
+is then simply created in the rebased `M` just like it was before the rebase.
+
+
+
+```
+$ jj log
+N rename baz->baz2, modify foo
+|
+| M rename bar->bar2, modify baz
+| |
+| | L rename foo->foo2, modify bar
+| |/
+|/
+K add foo, bar, baz
+$ jj new L M N
+```
+
+#### Example: Propagate changes to copied file, then rebase back
+
+```
+M foo="M"
+|
+| L copy foo->bar
+|/
+K add foo="K"
+```
+
+Let's say we rebase `M` onto `L`. If we decide to propagate changes to copies,
```

Comment:
I am not getting this example. Should this say "let's say we rebase L onto M" ?

Reply:
I think this is obsolete now with the change to say that we don't automatically propagate changes to copies